pandas/io/formats/latex.py

```diff
@@ -244,3 +245,74 @@ def _print_cline(self, buf, i, icol):
                           .format(cl=cl[1], icol=icol))
         # remove entries that have been written to buffer
         self.clinebuf = [x for x in self.clinebuf if x[0] != i]
+
+    def _write_tabular_begin(self, buf, column_format):
```

Comment:
The experts can comment on what they'd prefer, but since` _write_tabular_begin` and `_write_longtable_begin` are so similar I think you could try and merge them (same with the `*_end` functions (see [here](https://github.com/tomneep/pandas/commit/b4661e61b990d937a2f9968772a93c8968e7cf04) for a quick example, but careful that I changed a test without checking the latex output was the same)

Reply:
That may be good.

Another note, I'd prefer that the majority of the logic be implemented in a pure function that simply builds the string.  Then in the main function `write_result` we can do something like

```
buf.write(self._write_tabular_begin(column_format))
```

that makes testing a bit easier, as you can isolate the construction of the string to write from the actual IO.